core/config/routes.rb

```diff
@@ -5,6 +5,8 @@
     options[:host] = Spree.cdn_host if Spree.cdn_host.present?
     options[:host] ||= Rails.application.routes.default_url_options[:host]
 
+    options[:protocol] = 'https://' if Rails.env.production?
```

Comment:
@mad-eel 

1. I think a better check would be `Rails.application.config.force_ssl`
2. Protocol itself should be `https` without the `://`